jsapp/js/components/permissions/utils.ts

```diff
@@ -460,6 +461,66 @@ export function getPermLabel(perm: PermissionResponse) {
   return '???'
 }
 
+/**
+ * Returns the permission label with the correct suffix depending on the given asset type.
+ *
+ * Example: if we are sharing a library collection, the permissions will all say "[View, Edit, Manage] collection"
+ */
+export function getFriendlyPermLabelWithSuffix(
+  assetType: AssetTypeName | undefined,
+  perm?: PermissionResponse,
+  checkboxNameParam?: CheckboxNameAll,
+) {
+  let checkboxNameSuffix = ''
+  let checkboxNameAll = ''
+
+  if (perm) {
+    const permDef = permConfig.getPermission(perm.permission)
+
+    if (permDef) {
+      checkboxNameAll = getCheckboxNameByPermission(permDef.codename) || ''
+    }
+  }
+
+  if (checkboxNameParam) {
+    checkboxNameAll = checkboxNameParam
+  }
+
+  if (checkboxNameAll === 'formView' || checkboxNameAll === 'formEdit' || checkboxNameAll === 'formManage') {
+    switch (assetType) {
+      case AssetTypeName.block:
+        if (checkboxNameAll === 'formView') checkboxNameSuffix = t('View block')
+        if (checkboxNameAll === 'formEdit') checkboxNameSuffix = t('Edit block')
+        if (checkboxNameAll === 'formManage') checkboxNameSuffix = t('Manage block')
+        break
+      case AssetTypeName.collection:
+        if (checkboxNameAll === 'formView') checkboxNameSuffix = t('View collection')
+        if (checkboxNameAll === 'formEdit') checkboxNameSuffix = t('Edit collection')
+        if (checkboxNameAll === 'formManage') checkboxNameSuffix = t('Manage collection')
+        break
+      case AssetTypeName.template:
+        if (checkboxNameAll === 'formView') checkboxNameSuffix = t('View template')
+        if (checkboxNameAll === 'formEdit') checkboxNameSuffix = t('Edit template')
+        if (checkboxNameAll === 'formManage') checkboxNameSuffix = t('Manage template')
+        break
+      case AssetTypeName.question:
+        if (checkboxNameAll === 'formView') checkboxNameSuffix = t('View question')
+        if (checkboxNameAll === 'formEdit') checkboxNameSuffix = t('Edit question')
+        if (checkboxNameAll === 'formManage') checkboxNameSuffix = t('Manage question')
+        break
+      case AssetTypeName.survey:
+        if (checkboxNameAll === 'formManage') {
+          checkboxNameSuffix = 'project'
+        } else {
+          checkboxNameSuffix = 'form'
```

Comment:
And I am bit confused here… For `collection` etc. we are returning "View collection", but for `survey` we are returning "project" instead of "View project" (and "form" instead of "View form"?